[DOCS-2768] Expand and update Serilog setup steps (#12383)
DataDog/documentation: content/en/logs/log_collection/csharp.md

  text: "Log Collection Troubleshooting Guide"
---

> To send your C# logs to Datadog, we recommend [logging to a file][1] and then [configuring the Datadog Agent][2] to tail the file. See example logging configurations for [Serilog][3], [NLog][4], and [log4net][5]. Datadog recommends setting up your logging library to output logs in JSON format to avoid the need for [custom parsing rules][6].  

Reviewer: ```suggestion
To send your C# logs to Datadog, [log to a file][1] and then [configuring the Datadog Agent][2] to tail the file. See the example logging configurations for [Serilog][3], [NLog][4], and [log4net][5]. Datadog recommends setting up your logging library to output logs in JSON format to avoid the need for [custom parsing rules][6].  
```

{{% /tab %}}
{{< /tabs >}}

## Configure your Datadog Agent

> To start submitting logs to Datadog, once [log collection is enabled][10], set up [custom log collection][11] to tail the log file by doing the following:

Reviewer: ```suggestion
Once [log collection is enabled][10], set up [custom log collection][11] to tail your log files and send them to Datadog.
```

          #log_processing_rules:
          #  - type: multi_line
          #    name: new_log_start_with_date
          #    pattern: \d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])
      ```
3. [Restart the Agent][13]
4. Run the [Agent’s status subcommand][14] and look for `csharp` under the `Checks` section to confirm logs are successfully submitted to Datadog.

If logs are in JSON format, Datadog automatically [parses the log messages][15] to extract log attributes. Use the [Log Explorer][16] to view and troubleshoot your logs.

## Agentless logging

In cases where logs cannot be logged to a file because the application is running on an inaccessible host, logs can be sent directly from the application to Datadog. Outside of this specific use case, Datadog does not recommend this setup because handling connection issues should not be done directly in the application.

{{< tabs >}}
{{% tab "Serilog" %}}

1. To install the [Datadog Serilog sink][1], which sends events and logs to Datadog, run the following command in the Package Manager Console:

    ```text
    PM> Install-Package Serilog.Sinks.Datadog.Logs
    ```

> 2. Add the following code to initialize the logger directly in your application (do not forget to add your [API key][2]):

Reviewer: ```suggestion
2. Add the following code to initialize the logger directly in your application. Your [API key][2] is required for initialization.
```
